launch_testing/launch_testing/pytest/hooks.py

```diff
@@ -122,10 +125,22 @@ def reportinfo(self):
         return self.fspath, 0, 'launch tests: {}'.format(self.name)
 
 
-class LaunchTestModule(pytest.File):
+class LaunchTestModulePytest7(pytest.File):
 
-    def __init__(self, parent, *, fspath):
-        super().__init__(parent=parent, fspath=fspath)
```

Comment:
Seeing a warning in `LaunchROSTestModule`

```
./../../../../../../../usr/lib/python3/dist-packages/pydocstyle/config.py:6
  Warning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working

../../../../../../.local/lib/python3.8/site-packages/_pytest/nodes.py:140
../../../../../../.local/lib/python3.8/site-packages/_pytest/nodes.py:140
../../../../../../.local/lib/python3.8/site-packages/_pytest/nodes.py:140
  Warning: The (fspath: py.path.local) argument to LaunchROSTestModule is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
```

using

```
pip3 install pytest==7.0 && colcon test --event-handlers console_direct+ --packages-select ros2param
```

I think filtering the `fspath` argument if it's passed to the constructor may fix it as done here

https://github.com/ros2/launch/pull/592/files#diff-ae6811e267c65bf66fd7dfb5d9142aa58f70d41c354ba3c652078ee77ea2c271R140-R144

```python
       if _pytest_version_ge(7):
            if 'fspath' in kwargs:
                if kwargs['fspath'] is not None:
                    kwargs['path'] = pathlib.Path(kwargs['fspath'])
                del kwargs['fspath']
```